nightfall-deployer/contracts/KYC.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: CC0-1.0
+import '@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol';
+import './Ownable.sol';
+
+pragma solidity ^0.8.0;
+
+contract KYC is Initializable, Ownable {
+   
+    bool public whitelisting;
+    mapping(address => uint256) public users;
+    mapping(address => uint256) public managers;
+
+    // groupIds cannot be zero, although we don't specifcally chack for this because assigning a groupId of zero has
+    // no effect other than wasting gas.
+
+    function initialize() override virtual public initializer {
+        whitelisting = false;
+        Ownable.initialize();
+    }
+
+    function addUserToWhitelist(address _user) external {
+        // if a non-manager calls this, they will just assign someone to the zero group, which is the null value and has no effect
+        users[_user] = managers[msg.sender];
+    }
+
+    function removeUserFromWhitelist(address _user) external {
+        require(users[_user] != 0, 'This user is not whitelisted, so cannot be delisted');
+        require (managers[msg.sender] == users[_user], 'You are not the manager of this group' );
+        delete users[_user];
```

Comment:
From my little knowledge of Solidity: I see your comment but I wonder: wouldn't it be worth to check if the caller is manager? Or to have a modifier

Reply:
Hi @imagobea, I don't think these checks are needed.  For the `addUserToWhistlist` function, if the caller is not  a manager then the mapping `managers` will return a value of zero (the 'undefined' value for an integer type).  This will then add the user to the 'zero' group, which is the same as not being in a group.  This means that if a non-manager calls this function, it has no effect.
For the removeUserFromWhitelist function, if the caller is not a manager then the second `require` will revert because the managers mapping will return a value of zero but the users mapping cannot be zero because of the first `require`.